common/include/pcl/common/common.h

```diff
@@ -282,7 +282,12 @@ namespace pcl
     * \ingroup common
     */
   template<typename IteratorT, typename Functor> inline auto
-  computeMedian (IteratorT begin, IteratorT end, Functor f) noexcept -> typename std::result_of<Functor(decltype(*begin))>::type
+  computeMedian (IteratorT begin, IteratorT end, Functor f) noexcept ->
+  #if __cpp_lib_is_invocable
+  std::invoke_result_t<Functor(decltype(*begin))>
```

Comment:
```suggestion
  std::invoke_result_t<Functor, decltype(*begin)>
```
See https://en.cppreference.com/w/cpp/types/result_of